gcc/jit/jit-recording.h

```diff
@@ -1500,6 +1503,7 @@ class lvalue : public rvalue
   void set_alignment (unsigned bytes);
   unsigned get_alignment () const { return m_alignment; }
   virtual string * get_name () const { return NULL; }
+  virtual void set_name (const char *new_name) {}
```

Comment:
Should this be the following:
```suggestion
  virtual void set_name (const char *new_name) = 0?
```
instead?

Reply:
It doesn't compile because some children of `lvalue` don't have name and therefore don't implement `set_name` or `get_name`.

Reply:
So here are the types that do not implement `set_name` or `get_name`:

* `local`
* `access_field_of_lvalue`
* `dereference_rvalue`
* `dereference_field_rvalue`
* `vector_access`
* `array_access`

Reply:
Ah I see. Sure, let me check how to do it.